libraries/AP_NavEKF2/AP_NavEKF2_core.h

```diff
@@ -482,6 +485,26 @@ class NavEKF2_core
         bool            posReset;   // true when the position measurement has been reset
     };
 
+    /*
+      common intermediate variables used by all cores. These save a
+      lot memory by avoiding allocating these arrays on every core
+      Having these as stack variables would save even more memory, but
+      would give us very high stack usage in some functions, which
+      poses a risk of stack overflow until we have infrastructure in
+      place to calculate maximum stack usage using static analysis.
+     */
+    struct core_common {
```

Comment:
A comment here indicating we're making the assumption everything in this structure is a float might be a good idea.

Reply:
done